clang/runtime/dpct-rt/include/dpct/graph.hpp

```diff
@@ -133,5 +153,66 @@ static void add_dependencies(dpct::experimental::command_graph_ptr graph,
   }
 }
 
+/// \Queries the nodes present in graph 
+/// \param [in] graph A pointer to the command graph.
+/// \param [in] numberOfNodes number of Nodes in graph.
+
+inline std::vector<sycl::ext::oneapi::experimental::node> get_nodes(
+    const dpct::experimental::command_graph_ptr graph,
+    std::size_t numberOfNodes) {
+
+    return detail::get_nodes_impl(std::function<std::vector<sycl::ext::oneapi::experimental::node>()>(
+        [&]() { return graph->get_nodes(); }), numberOfNodes);
+    ;
+}
+
+
+/// \Queries the root nodes present in graph 
+/// \param [in] graph A pointer to the command graph.
+/// \param [in] numberOfRootNodes number of root nodes in graph.
+inline std::vector<sycl::ext::oneapi::experimental::node> get_root_nodes(
+    const dpct::experimental::command_graph_ptr graph,
+    std::size_t numberOfRootNodes) {
+
+    return detail::get_nodes_impl(std::function<std::vector<sycl::ext::oneapi::experimental::node>()>(
+        [&]() { return graph->get_root_nodes(); }), numberOfRootNodes);
+    ;
+}
+
+/// \Queries the successors present in graph node 
+/// \param [in] graph A pointer to the command graph.
+/// \param [in] numberOfSuccessorNodes number of successor nodes in graph.
+inline std::vector<sycl::ext::oneapi::experimental::node> get_root_nodes(
+    const dpct::experimental::node_ptr node,
+    std::size_t numberOfSuccessorNodes) {
+
+    return detail::get_nodes_impl(std::function<std::vector<sycl::ext::oneapi::experimental::node>()>(
+        [&]() { return node->get_successors(); }), numberOfSuccessorNodes);
+    ;
+}
+
+/// \Queries the predecessors present in graph node
+/// \param [in] graph A pointer to the command graph.
+/// \param [in] numberOfPredecessorNodes number of predecessor nodes in graph.
+inline std::vector<sycl::ext::oneapi::experimental::node> get_predecessors(
+    const dpct::experimental::node_ptr node,
+    std::size_t numberOfPredecessorNodes) {
+ 
+    return detail::get_nodes_impl(std::function<std::vector<sycl::ext::oneapi::experimental::node>()>(
+        [&]() { return node->get_predecessors(); }), numberOfPredecessorNodes);
+    
+}
+
+/// \Prints the graph 
+/// \param [in] graph A pointer to the command graph.
+/// \param [in] path FilePath Path to the output file.
+/// \param [in] verbose If true, print additional information about the nodes such
+/// as kernel args or memory access where applicable.
+static inline void print_graph(const dpct::experimental::command_graph_ptr graph,
```

Comment:
Do we have other reason for adding this helper function?
Can we migrate the code to `graph->print_graph(path, verbose)` directly?